dataflow/src/main/java/org/checkerframework/dataflow/analysis/AbstractAnalysis.java

```diff
@@ -210,6 +211,176 @@ public AnalysisResult<V, S> getResult() {
     return nodeValues.get(n);
   }
 
+  /**
+   * Returns the store immediately before a given {@link Tree}. Only intended for use while the
+   * analysis is running.
+   *
+   * @param tree a tree
+   * @param analysisCaches caches of analysis results. If it is not null, this method uses and
+   *     updates it. It is a map from a TransferInput for a Block to a map. The inner map is from a
+   *     node within the block to a TransferResult.
+   * @return the store immediately before a given {@link Tree}
+   */
+  public @Nullable S getStoreBefore(
+      Tree tree,
+      @Nullable Map<TransferInput<V, S>, IdentityHashMap<Node, TransferResult<V, S>>>
+          analysisCaches) {
+    if (!isRunning()) {
+      throw new BugInCF(
+          "AbstractAnalysis::getStoreBefore() should only be called when the analysis is running.");
+    }
+    Set<Node> nodes = getNodesForTree(tree);
+    if (nodes == null) {
+      return null;
+    }
+    return getStoreBefore(nodes, analysisCaches);
+  }
+
+  /**
+   * Returns the store immediately before a given set of {@link Node}s. Only intended for use while
```

Comment:
I think this isn't your fault because such a comment pre-existed you, but the wording "the store" implies that one exists that is before all the nodes.  In fact, this returns a new merged store that is valid for each of the nodes, even if not maximally accurate for any of them.